### ISSUE TITLE
Proposal: dotenv + computed env types in std

### ISSUE THREAD

**Issue by GerbenRampaart** [denoland/deno#13396]
Part of almost every node project I've done is validating the env variables, both their existence and content (type, formatting, defaults and such). Joi is a great way to do that, I've also created my own module as a simpler version of what Joi does (https://www.npmjs.com/package/dotenv-typed) and for Deno there is the computed_types third party (https://deno.land/x/computed_types@v1.1.0).

My proposal is to merge computed_types (not my project though) into std. And, for that matter, I feel dotenv should be embedded in std because the .env has become such a mainstream way to solve the development environment env problem.

In summary: extend std with dotenv with rich type and value checking to be added to Deno.env

What do you think? Thank you.

**Comment by GerbenRampaart** [comment]
To slightly elaborate in case anyone wonders why the strict checking on env variables; I work in a corporate environment where key/value stores are managed by technical application managers who monitor and supports apps in production (like values.yml for kubernetes). They frequently enter wrong values, or make typos. I imagine this happens for many other developers as well. Clean errors in the logs and thus easy strict env checking is really helpful.

**Comment by bartlomieju** [comment]
I'm not familiar with either `joi` or `dotenb-typed`; but having `dotenv` as a `deno_std` module makes sense to me. If the author of that package would be willing to have it merged in `deno_std` we can discuss this topic.

**Comment by pietvanzoen** [comment]
I think it’s a solid idea. I don’t know a project these days that doesn’t use dotenv of some sort. 

Let me know what needs to happen. Happy to assist.

**Comment by bartlomieju** [comment]
@pietvanzoen thanks! If you are willing to move `dotenv` into `deno_std` then please open a PR. I went over the README in https://deno.land/x/dotenv and the API seems very reasonable, so I don't expect any changes would be necessary besides formatting/linting.

**Comment by pietvanzoen** [comment]
Hey @bartlomieju @GerbenRampaart. Apologies for the delay in this. I've made the PR here: https://github.com/denoland/deno_std/pull/1877

**Comment by roj1512** [comment]
@kt3k This issue should be reopened. A big part of it is not resolved: "computed env types".

**Comment by GerbenRampaart** [comment]
Maybe computed env types was overstretching the scope of this issue. dotenv was useful enough to be the focus here. Plus when it comes to computed types in my experience there are lots of ways to solve that problem so it's probably harder to find consensus for adding that to std.

I wrote a lightweight npm for node to solve that years ago (https://www.npmjs.com/package/dotenv-typed) but Joi is a bigger, heavier and way more popular approach to computed types.

Also, there are computed types solutions already for deno, like https://deno.land/x/computed_types@v1.11.2

Though really, I'm not sure if std is then not just 'doing too much'. It might feel like a bit of a bloat. Don't know how the deno team feels.

tldr: dotenv is so common in the node/deno world that it was justified to standardise in std, computed types maybe not (imho).
